components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -311,6 +311,11 @@ void AdsServiceImpl::OnCreate() {
   }
 }
 
+void AdsServiceImpl::DisableAds() {
+  set_ads_enabled(false);
```

Comment:
calling set_ads_enabled(false) already calls Stop

Reply:
Yep. Missed the PrefChanged event.